keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,525 @@
+        return getPagedSecrets(service.getSecretVersions(vaultEndPoint, name, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * List secrets in a specified key vault.
+     * The list secrets operation is applicable to the entire vault.
+     * However, only the base secret identifier and its attributes are provided in the response. Individual secret versions are not listed in the response.
+     * This operation requires the secrets/list permission.
+     *
+     * @param maxPageResults Maximum number of results to return in a page. If not specified, the service will return up to 25 results.
+     * @throws NullPointerException thrown if maxPageResults parameter is null
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the secrets in the key vault.
+     */
+    public Flux<SecretAttributes> listSecretsAsync(int maxPageResults) {
+        Objects.requireNonNull(maxPageResults, "The maximum page results parameter cannot be null.");
+        return getPagedSecrets(service.getSecrets(vaultEndPoint, maxPageResults, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * Lists {@link DeletedSecret} secrets of the key vault.
+     * The list deleted secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @param maxPageResults Maximum number of results to return in a page. If not specified, the service will return up to 25 results.
+     * @throws NullPointerException thrown if maxPageResults parameter is null
+     * @return A {@link Flux} containing all of the {@link DeletedSecret deleted secrets} in the vault.
+     */
+    public Flux<DeletedSecret> listDeletedSecretsAsync(int maxPageResults) {
+        Objects.requireNonNull(maxPageResults, "The maximum page results parameter cannot be null.");
+        return getPagedDeletedSecrets(service.getDeletedSecrets(vaultEndPoint, maxPageResults, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * List all versions of the specified secret.
+     * The full secret identifier and attributes are provided in the response. No values are returned for the secrets.
+     * This operations requires the secrets/list permission.
+     *
+     * @param name The name of the secret.
+     * @param maxPageResults Maximum number of results to return in a page. If not specified, the service will return up to 25 results.
+     * @throws NullPointerException thrown if {@code name} or {@code maxPageResults} is null
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the versions of the secret in the key vault. Flux is empty if secret with {@code name} does not exist in key vault
+     */
+    public Flux<SecretAttributes> listSecretVersionsAsync(String name, int maxPageResults) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
```

Comment:
Jonathan pointed out before that we probably want a string.isNullOrEmpty() check in commons library. It's possible to submit a name = "" from this code.

Reply:
Yep, there is a `ImplUtils` class in azure-common you should add this utility API into.

Reply:
Okay.
Temporarily, added empty string check here with TODO to update this, once we update azure common. (issue referenced)
Opened an issue to update ImplUtils in Azure Common:
Link:  Azure/azure-sdk-for-java#3373